core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -296,15 +290,19 @@ struct CudaParallelLaunch<
   }
 
   static cudaFuncAttributes get_cuda_func_attributes() {
-    static cudaFuncAttributes attr;
-    static bool attr_set = false;
-    if (!attr_set) {
+    // Race condition inside of cudaFuncGetAttributes if the same address is
```

Comment:
Should we mark it as a workaround for CUDA 11?  Is that something we are going to report upstream?

Reply:
I don't think its in principle a workaround just for CUDA 11. There is no implementation guarantee for the function, its just bad API design to take pointers to stuff instead of just returning the struct.